crates/storage/errors/src/writer.rs

```diff
@@ -15,17 +15,7 @@ pub enum UnifiedStorageWriterError {
     #[display("Static file writer is of wrong segment: got {_0}, expected {_1}")]
     IncorrectStaticFileWriter(StaticFileSegment, StaticFileSegment),
     /// Database-related errors.
-    Database(DatabaseError),
-}
-
-#[cfg(feature = "std")]
-impl std::error::Error for UnifiedStorageWriterError {
-    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
-        match self {
-            Self::Database(source) => std::error::Error::source(source),
-            _ => Option::None,
-        }
-    }
+    Database(#[error(source)] DatabaseError),
```

Comment:
does it need `error(source)` attribute? its a 1-ary tuple struct

Reply:
you can look at cargo expand to make sure

Reply:
fixed